src/main/java/org/hl7/davinci/priorauth/ClaimEndpoint.java

```diff
@@ -599,7 +606,36 @@ private Bundle generateAndStoreClaimResponse(Bundle bundle, Claim claim, String
 
     @Override
     public void run() {
-      updateClaim(bundle, claimId, patient, disposition);
+      if (updateClaim(bundle, claimId, patient, disposition) != null) {
+        // Check for subscription
+        Map<String, Object> constraintMap = new HashMap<String, Object>();
+        constraintMap.put("claimResponseId", claimId);
+        constraintMap.put("patient", patient);
+        Subscription subscription = (Subscription) App.getDB().read(Database.SUBSCRIPTION, constraintMap);
```

Comment:
Should we delete them after they are notified and it goes to a final state?

Reply:
No this is the responsibility of the client

> If the retrieved ClaimResponse has an outcome of ‘complete’ or ‘error’, the client SHALL perform a DELETE on the Subscription.

The client I made does this too FYI

Reply:
Sorry I missed it.